### DISCUSSION TITLE
Is KIND2 supports bitwise operators?

### DISCUSSION THREAD

**Discussion by kumori123** [kind2-mc/kind2#1000] [Q&A] [answered]
Hi, I tried to input this code
```
function f1(i1: uint8; i2: uint8) returns (o1: uint8)
let
    o1 = i1 | i2;
tel
```
to KIND2 ，but it seems like KIND2 cannot parse it correctly. 
So currently can KIND2 support those kinds of code above?
Thank you!

**Comment by daniel-larraz** [comment, marked as answer]
Yes, Kind 2 supports bit-wise operators. You can find the list of supported operators in the [documentation](https://kind.cs.uiowa.edu/kind2_user_docs/v2.0.0/2_input/3_machine_ints.html).

The correct syntax for your example is:
```
function f1(i1: uint8; i2: uint8) returns (o1: uint8)
let
    o1 = i1 || i2;
tel
```

**Reply by kumori123** [reply]
Thank you!
